release: test_python misses protobuf dependency

```text
=== Testing adbc_driver_flightsql ===
env python -m pytest -vv /tmp/arrow-adbc-0.8.0.OUbjD/apache-arrow-adbc-0.8.0/python/adbc_driver_flightsql/tests
============================= test session starts ==============================
platform linux -- Python 3.11.5, pytest-7.4.3, pluggy-1.3.0 -- /tmp/arrow-adbc-0.8.0.OUbjD/venv-source/bin/python
cachedir: .pytest_cache
rootdir: /tmp/arrow-adbc-0.8.0.OUbjD/apache-arrow-adbc-0.8.0/python/adbc_driver_flightsql
collecting ... collected 8 items / 1 error

==================================== ERRORS ====================================
____________________ ERROR collecting tests/test_errors.py _____________________
ImportError while importing test module '/tmp/arrow-adbc-0.8.0.OUbjD/apache-arrow-adbc-0.8.0/python/adbc_driver_flightsql/tests/test_errors.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../venv-source/lib/python3.11/site-packages/_pytest/python.py:617: in _importtestmodule
    mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
../venv-source/lib/python3.11/site-packages/_pytest/pathlib.py:567: in import_path
    importlib.import_module(module_name)
/usr/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1204: in _gcd_import
    ???
<frozen importlib._bootstrap>:1176: in _find_and_load
    ???
<frozen importlib._bootstrap>:1147: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:690: in _load_unlocked
    ???
../venv-source/lib/python3.11/site-packages/_pytest/assertion/rewrite.py:186: in exec_module
    exec(co, module.__dict__)
python/adbc_driver_flightsql/tests/test_errors.py:20: in <module>
    import google.protobuf.any_pb2 as any_pb2
E   ModuleNotFoundError: No module named 'google'
=========================== short test summary info ============================
ERROR python/adbc_driver_flightsql/tests/test_errors.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.12s ===============================
```

--- a/dev/release/verify-release-candidate.sh
+++ b/dev/release/verify-release-candidate.sh
@@ -456,7 +456,7 @@ test_python() {
   show_header "Build and test Python libraries"
 
   # Build and test Python
-  maybe_setup_virtualenv cython duckdb pandas pyarrow pytest setuptools_scm setuptools || exit 1
+  maybe_setup_virtualenv cython duckdb pandas protobuf pyarrow pytest setuptools_scm setuptools || exit 1
   maybe_setup_conda --file "${ADBC_DIR}/ci/conda_env_python.txt" || exit 1
 
   if [ "${USE_CONDA}" -gt 0 ]; then

